src/MdOpts.cpp

```diff
@@ -0,0 +1,69 @@
+#include "MdOpts.h"
+#include "ArgList.h"
+#include "CpptrajStdio.h"
+#include "Constants.h"
+
+MdOpts::MdOpts() :
+  shakeType_(Constraints::OFF),
+  scaleEE_(1.0/1.2), // Amber default
+  scaleNB_(1.0/2.0), // Amber default
+  cutEE_(8.0),       // in Ang., Amber default
+  cutNB_(8.0),       // in Ang., Amber default
+  qfac_(Constants::ELECTOAMBER * Constants::ELECTOAMBER), // Assume charges in elec. units, Amber default
+  nExclude_(4)       // Exclude dihedral, angle, bond
+{}
+
+/** Display options recognized by MdOpts to stdout. */
+void MdOpts::PrintHelp() {
+  mprintf("\tcut <cutoff>     : Set nonbonded interaction cutoff in Ang. (electrostatics and vdW).\n"
+          "\tcutee <cutoff>   : Set electrostatics interaction cutoff in Ang.\n"
+          "\tcutnb <cutoff>   : Set vdW interaction cutoff in Ang.\n"
+          "\tscaleee <factor> : Scaling factor to multiply 1-4 electrostatics interactions by.\n"
+          "\tscalenb <factor> : Scaling factor to multiply 1-4 vdW interactions by.\n"
+          "\tshake <type>     : Use SHAKE constraints of <type>;\n"
+          "\t                   <type> = 'hydrogen' : Constrain bonds to hydrogen.\n"
+          "\t                   <type> = 'all'      : Constrain all bonds.\n"
+          "\tnexclude <#>     : Number of bonded atoms within which nonbonded interactions are excluded.\n"
+          "\tqfac <factor>    : Factor to use in electrostatic calculation.\n"
+  );
+}
+
+/** Get options from incoming argument list. */
+int MdOpts::GetOptsFromArgs(ArgList& argIn)
+{
+  cutEE_ = argIn.getKeyDouble("cutee", 8.0);
+  cutNB_ = argIn.getKeyDouble("cutnb", 8.0);
+  if (argIn.Contains("cut")) {
+    double cut = argIn.getKeyDouble("cut", 8.0);
+    cutEE_ = cut;
+    cutNB_ = cut;
+  }
+  scaleEE_ = argIn.getKeyDouble("scaleee", 1.0/1.2);
+  scaleNB_ = argIn.getKeyDouble("scalenb", 1.0/2.0);
+  shakeType_ = Constraints::OFF;
+  std::string shakearg = argIn.GetStringKey("shake");
+  if (!shakearg.empty()) {
+    if (shakearg == "h" || shakearg == "hydrogen")
+      shakeType_ = Constraints::BONDS_TO_H;
+    else if (shakearg == "all")
+      shakeType_ = Constraints::ALL_BONDS;
```

Comment:
I think openmm also supports h angles if that's worth supporting here.

Reply:
Cool - eventually I would like to add support for all features supported by OpenMM - just wanted to get the initial stuff in the master branch so things don't diverge too much.